src/git_changelog/build.py

```diff
@@ -25,6 +25,59 @@
 ConventionType = Union[str, CommitConvention, Type[CommitConvention]]
 
 
+class PrefixedVersionInfo:
+    """A VersionInfo with an optional prefix."""
+
+    def __init__(self, prefix: str, version: VersionInfo):
+        """
+        Initialization method.
+
+        Arguments:
+            prefix: The (optional) version prefix.
+            version: The wrapped version.
+
+        """
+        self._prefix = prefix
+        self._version = version
+
+    def __str__(self):
+        return self._prefix + str(self._version)
+
+    @classmethod
+    def parse(cls, version: str) -> PrefixedVersionInfo:
+        """
+        Parse a string representing a version with an optional prefix.
+
+        Arguments:
+            version: The string to parse.
+
+        Returns:
+            The parsed version.
+        """
+        prefix = ""
+        if version[0] == "v":
+            prefix = "v"
+            version = version[1:]
+
+        return cls(prefix, VersionInfo.parse(version))
+
+    def __getattr__(self, attr):
+        def encapsulate(obj):  # noqa: WPS430
+            if isinstance(obj, VersionInfo):
+                klass = type(self)
+                obj = klass(self._prefix, obj)
+            return obj
+
+        ret = getattr(self._version, attr)
+        if callable(ret):
+
+            def wrapper(*args, **kwargs):  # noqa: WPS430
+                return encapsulate(ret(*args, **kwargs))
+
+            return wrapper
+        return encapsulate(ret)
```

Comment:
That's rather complex. I'm not I want to maintain that. I also don't see the usefulness of the class :confused: